compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -348,6 +348,8 @@ impl<'context> Elaborator<'context> {
                 let func_type =
                     self.type_check_variable(function_name, function_id, turbofish_generics);
 
+                self.interner.push_expr_type(function_id, func_type.clone());
```

Comment:
This is an unrelated one line fix I forgot about. We weren't pushing a function identifier's type before after desugaring a method call into a function call.